File: Math/1390. Four Divisors.py
class Solution:
    def sumFourDivisors(self, nums: List[int]) -> int:
        def divisor(num):
            store = set()
            i, sum = 1, 0
            while i * i <= num:
                if not num % i:
                    store.add(i)
                    store.add(num // i)
                i += 1
            if len(store) == 4:
                for div in store:
                    sum += div
            return sum
        
        ans = 0
        count = Counter(nums)
        for num in count:
            ans += divisor(num) * count[num]
        return ans

File: Math/1390. Four Divisors.cpp
class Solution {
public:
    void divisor(int n, int &sum)
    {
        vector<int> arr;
        for(int i = 1; i * i <= n; i++)
        {
            if(n % i == 0)
            {
                arr.push_back(i);
                if(n / i != i)
                    arr.push_back(n / i);
            }
        }
        if (arr.size() == 4)
            for (auto val : arr)
                sum += val;
    }
    int sumFourDivisors(vector<int> &nums)
    {
        int sum = 0;
        for(auto val : nums)
            divisor(val, sum);
        return sum;
    }
};